doc/source/dev/contributor/debugging_linalg_issues.rst

```diff
@@ -477,4 +477,63 @@ file to automate this and avoid the manual paths:
    machine, it's easy to accidentally pick up the wrong one during the build
    (remember: ``-lopenblas`` only says "link against *some*
    ``libopenblas.so``). If you're not sure, use ``ldd`` on the test executable
-   you built to inspect which shared library it's linked again.
+   you built to inspect which shared library it's linked against.
+
+
+Debugging linalg issues with ``gdb``
+------------------------------------
+
+When debugging linalg issues, it is sometimes useful to step through both Python
+and C code. You can use ``pdb`` for the former, and ``gdb`` for the latter.
+
+First, prepare a small python reproducer, with a breakpoint. For example::
+
+    $ cat chol.py
+    import numpy as np
+    from scipy import linalg
+    n = 40
+    rng=np.random.refault_rng(1234)
+    x = rng.uniform(size=n)
+    a = x[:, None] @ x[None, :] + np.identity(n)
+
+    breakpoint()      # note a breakpoint
+    linalg.cholesky(a)
+
+Then, you will need to run it under the ``gdb`` and add a C-level breakpoint at
+the LAPACK function. This way, your execution will stop twice: first on the
+Python breakpoint and then on the C breakpoint, and you will be able to step
+through and inspect values of both Python and C variables.
+
+To find out the LAPACK name, read the python source of the SciPy function and
+use ``nm`` on the ``.so`` library to find out the exact name.
+For the Cholesky factorization above, the LAPACK function is ``?potrf``, and the
+C name on Ubuntu linux is ``dpotrf_`` (it may be spelled with or without the
+trailing underscore, in uppper case or lower case, depending on the system).
+
+Here is an example ``gdb`` session::
+
+    $ gdb --args python chol.py
+    ...
+    (gdb) b dpotrf_     # this adds a C breakpoint (type "y" below)
+    Function "dpotrf_" not defined.
+    Make breakpoint pending on future shared library load? (y or [n]) y
+    Breakpoint 1 (dpotrf_) pending.
+    (gdb) run    # run the python script
+    ...
+    > /home/br/temp/chol/chol.py(12)<module>()
+    -> linalg.cholesky(a)   # execution stopped at the python breakpoint
+    (Pdb)   # ... inspect the python state here
+    (Pdb) c     # continue execution until the C breakpoint
+
+    Thread 1 "python" hit Breakpoint 1, 0x00007ffff4c48820 in dpotrf_ ()
+       from /home/br/mambaforge/envs/scipy-dev/lib/python3.10/site-packages/numpy/core/../../../../libcblas.so.3
+    (gdb) s     # step through the C function
+    Single stepping until exit from function dpotrf_,
+    which has no line number information.
+    f2py_rout__flapack_dpotrf (capi_self=<optimized out>, capi_args=<optimized out>, 
+        capi_keywds=<optimized out>, f2py_func=0x7ffff4c48820 <dpotrf_>)
+        at scipy/linalg/_flapackmodule.c:63281
+    ....
+    (gdb) p lda    # inspect values of C variables etc
```

Comment:
I get:
```
(gdb) p lda
No symbol "lda" in current context.
```
also for most other operations:
```
(gdb) info args
No symbol table info available.
```

It may be useful to extend this comment to say that this may not work. And/or add an example using of `bt`, which is probably the thing that always works (just may show some missing symbols:
```
(gdb) bt
#0  0x00007ffff3056b1e in f2py_rout.flapack_dpotrf ()
   from /path/to/site-packages/scipy/linalg/_flapack.cpython-311-x86_64-linux-gnu.so
#1  0x0000555555734323 in _PyObject_MakeTpCall (tstate=0x555555ad0558 <_PyRuntime+166328>, 
    callable=<fortran at remote 0x7ffff40ffc00>, args=<optimized out>, nargs=1, keywords=('lower', 'overwrite_a', 'clean'))
    at /usr/local/src/conda/python-3.11.8/Objects/call.c:214
```

Reply:
Hmmm, TBH I've no idea why symbols are not available and how to fix this. Changing the build type to "debug" in meson.build does not seem to change anything, at least locally.
If you can suggest a magic meson incantation, am happy to try it out.
Otherwise, the updated version adds a comment and mentions `bt`.

Reply:
Okay, I got tricked by conda's environment activation again. It's important to build with `debug` build type, _and_ unset `CFLAGS`/`CXXFLAGS`. Then everything works as expected. You could note that in a single sentence above the example, and link to https://numpy.org/devdocs/dev/development_environment.html#debugging

Reply:
In fact, here's a question about `no symbol in current context`. Did you actually step into the function? Here's what I see locally:

```
-> linalg.cholesky(a)   # segfaults
(Pdb) c            # <<< run until the gdb breakpoint

Thread 1 "python" hit Breakpoint 1, 0x00007ffff4c48820 in dpotrf_ ()
   from /home/br/mambaforge/envs/scipy-dev/lib/python3.10/site-packages/numpy/core/../../../../libcblas.so.3
(gdb) p lda                                              # <<< oops
No symbol "lda" in current context.
(gdb) info args
No symbol table info available.
(gdb) s                                                    # <<<< step in
Single stepping until exit from function dpotrf_,
which has no line number information.
f2py_rout__flapack_dpotrf (capi_self=<optimized out>, capi_args=<optimized out>, 
    capi_keywds=<optimized out>, f2py_func=0x7ffff4c48820 <dpotrf_>)
    at scipy/linalg/_flapackmodule.c:63281
63281	                (*f2py_func)((lower?"L":"U"),&n,a,&lda,&info); if(clean){Py_ssize_t i,j;if(lower){for(i=0;i<n;++i) for(j=i+1;j<n;++j) *(a+j*n+i)=0.0;} else {for(i=0;i<n;++i) for(j=i+1;j<n;++j) *(a+i*n+j)=0.0;}} ;
(gdb) info args                                  # <<<<< symbol table is available
capi_self = <optimized out>
capi_args = <optimized out>
capi_keywds = <optimized out>
f2py_func = 0x7ffff4c48820 <dpotrf_>
```

Reply:
Fairly sure I did, but not 100% sure. Can't check right now, I'm on another machine.

Reply:
> with debug build type, and unset CFLAGS/CXXFLAGS. Then everything works as expected. You could note that in a single sentence above the example, and link to https://numpy.org/devdocs/dev/development_environment.html#debugging

Updated to the suggestion. 
For me locally unsetting CFLAGS and CXXFLAGS makes no difference, but hey.

Thanks for the NumPy link! I'm pretty sure I saw that at some point, and then forgot. In principle, this PR is at best a small addition on top of that one. From where I stand, this addition is writing out what I found helpful, while it's still fresh in memory. We can either assume there is at least one other person who would find this helpful, or assume the numpy version is sufficient and reject this PR. Reviewer's choice :-).

Reply:
I'm happy for it to go in. The additional detail and the BLAS-specific notes on symbol naming will be helpful for some contributors I think.